File: competitive_programming/codeforces/11_golang/977a_wrong_subtraction.go
package main

import "fmt"

func wrongSubtraction(number int64, steps int) int64 {
	for i := 0; i < steps; i++ {
		if number%10 == 0 {
			number /= 10
		} else {
			number--
		}
	}
	return number
}

func main() {
	if wrongSubtraction(512, 4) != 50 || wrongSubtraction(1000000000, 9) != 1 {
		panic("wrong subtraction failed")
	}
	fmt.Println("977A wrong subtraction ok")
}

File: competitive_programming/codeforces/07_typescript/977a_wrong_subtraction.ts
function wrongSubtraction(number: number, steps: number): number {
  let current = number;
  for (let i = 0; i < steps; i += 1) {
    if (current % 10 === 0) current /= 10;
    else current -= 1;
  }
  return current;
}

if (wrongSubtraction(512, 4) !== 50) throw new Error("wrong subtraction failed");
if (wrongSubtraction(1000000000, 9) !== 1) throw new Error("wrong subtraction failed");
console.log("977A wrong subtraction ok");
